src/Congo.WebApi/Controllers/CartController.cs

```diff
@@ -16,12 +18,18 @@ public class CartController : ControllerBase
 
         public CartController(IMediator mediator) => _mediator = mediator;
 
-
         [HttpGet("{id}")]
         public async Task<ActionResult<CartResponse>> Get(Guid id)
         {
             var cart = await _mediator.Send(new GetCartQuery(id));
             return Ok(cart.Adapt<CartResponse>());
         }
+
+        [HttpPost("{cartId}")]
+        public async Task<ActionResult<Cart>> Post(Guid? cartId, Product product)
```

Comment:
This should take in a `product id` and a `quantity` and _optionally_ a `cart id` then it should return a new `cart id` if one **isn't** provided or `NoContent` if a cart id is provided. You can change signature to just return `Task<ActionResult>` then I think you can use attributes to tell swagger what this action actually returns.

Reply:
We will probably have multiple post actions here, so `Post` isn't the best name for this.

Reply:
Hey i'm a little unsure what you mean but using attributes to tell swagger what it actually returns

Reply:
We are using Swagger UI here, which shows you all your endpoints. This action may return 1 of 2 things, NoContent (204) if we supply a cart id or 200 with the cart id if we do not supply a cart id. So we want to let swagger know that this may return different types so they can show up here on the UI: 
![image](https://user-images.githubusercontent.com/15176357/143070916-aa0440f0-a53f-4f7f-bad0-c9ba1e636746.png)
We can do this by adding `ProducesResponseType` attribute to our action, like this:
![image](https://user-images.githubusercontent.com/15176357/143070976-470cffa9-83d7-4ec1-9143-099ab5575446.png)

Will also need to change the method signature like I did too.

One more thing, you will need to add another `HttpPost` attribute without the cartId, otherwise we will not be able to post without a cart id.